internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -385,43 +353,79 @@ func (c *ClusterManager) reconcileReadyCluster(cluster api.Cluster) error {
+		err = c.reconcileClusterInstanceType(cluster)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to reconcile instance type ready cluster %q", cluster.ClusterID)
+		}
+
+		err = c.reconcileDynamicCapacityInfo(cluster)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to reconcile dynamic capacity info for ready cluster %q", cluster.ClusterID)
+		}
+
+		err = c.reconcileClusterDNS(cluster)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to reconcile cluster dns of ready cluster %q", cluster.ClusterID)
+		}
+
+		err = c.reconcileClusterIdentityProvider(cluster)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to reconcile identity provider of ready cluster %q", cluster.ClusterID)
+		}
+
+		err = c.reconcileKasFleetshardOperator(cluster)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to reconcile Kas Fleetshard Operator of ready cluster %q:", cluster.ClusterID)
+		}
 	}
 
-	// resources update if needed
-	if err := c.reconcileClusterResources(cluster); err != nil {
-		return errors.WithMessagef(err, "failed to reconcile ready cluster resources %s ", cluster.ClusterID)
+	return nil
+}
+
+// reconcileClusterBaseKafkasDomainName sets the base domain name of deployed kafkas from the configuration if it is not already set.
+func (c *ClusterManager) reconcileClusterBaseKafkasDomainName(cluster api.Cluster) error {
+	logger.Logger.Infof("reconciling cluster = %s kafka domain name", cluster.ClusterID)
+	if !c.KafkaTLSCertificateManagementService.IsAutomaticCertificateManagementEnabled() {
+		logger.Logger.Infof("certificate management running in manual mode. Skipping reconciling base domain for kafkas for cluster %q", cluster.ClusterID)
+		return nil
 	}
 
-	err = c.reconcileClusterDNS(cluster)
-	if err != nil {
-		return errors.WithMessagef(err, "failed to reconcile cluster dns of ready cluster %s: %s", cluster.ClusterID, err.Error())
+	if cluster.BaseKafkasDomainName != "" {
+		logger.Logger.Infof("kafka domain name for cluster %q already set", cluster.ClusterID)
+		return nil
 	}
 
-	err = c.reconcileClusterIdentityProvider(cluster)
-	if err != nil {
-		return errors.WithMessagef(err, "failed to reconcile identity provider of ready cluster %s: %s", cluster.ClusterID, err.Error())
+	// we want to continue using the shared wildcard certificate under the kafka domain for already existing ready clusters.
+	// See ADR-89 https://github.com/bf2fc6cc711aee1a0c2a/architecture/blob/main/_adr/89/index.adoc for more info
+	if cluster.Status == api.ClusterReady && cluster.ClusterType == api.ManagedDataPlaneClusterType.String() {
+		cluster.BaseKafkasDomainName = c.KafkaConfig.KafkaDomainName
+	} else {
+		// use the domain format provided by ADR-89 https://github.com/bf2fc6cc711aee1a0c2a/architecture/blob/main/_adr/89/index.adoc
+		cluster.BaseKafkasDomainName = fmt.Sprintf("%s.%s", cluster.ClusterID, c.KafkaConfig.KafkaDomainName)
 	}
```

Comment:
@emmanuelbernard raised a comment https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/84#discussion_r1085235508 about using the cluster-id in the URL.

He'll raise it to the architecture call and depending on the outcome we might need to revisit this e.g using a unique generated string per cluster.

@emmanuelbernard can you confirm with us the path forward once we've the outcome of the architecture discussion?

/cc  @pb82

Reply:
a decision was reached to create a certificate per kafka instance.

I've opened https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1531 to take care of that.